### ISSUE TITLE
Missing functionality

### ISSUE THREAD

**Issue by DevDevious** [sarulabs/dingo#19]
Hello,

Currently missing in **sarulabs/dingo** :

`By default the objects are shared that means what they stored as singletons in the Container. You will retrieve the exact same object every time you call the Get method on the same Container. The Build function will only be called once. In case when you don't want the objects to be shared set the Unshared property of object definition to true.`

**dingo** service definitions do not have an `Unshared` property:

``` go

package dingo

// Def is the structure containing a service definition.
type Def struct {
	Name  string
	Scope string
	// NotForAutoFill should be set to true if you
	// do not want to use this service automatically
	// as a dependency in other services.
	NotForAutoFill bool
	// Build defines the service constructor. It can be either:
	// - a pointer to a structure: (*MyStruct)(nil)
	// - a factory function: func(any, any, ...) (any, error)
	Build interface{}
	// Params are used to assist the service constructor.
	Params Params
	// Close should be a function: func(any) error.
	// With any being the type of the service.
	Close interface{}
}
```

Any plans on adding this?

**Comment by sarulabs** [comment]
Hello,
I added the `Unshared` field on `master`.
I'm waiting for the issue #22 to be resolved to create new tags.

**Comment by DevDevious** [comment]
@sarulabs awesome! Thank you!

**Comment by sarulabs** [comment]
Available in `v4.1.0`.
